go/pkg/client/bytestream.go

```diff
@@ -15,19 +15,21 @@ import (
 )
 
 // WriteBytes uploads a byte slice.
-func (c *Client) WriteBytes(ctx context.Context, name string, data []byte) error {
+func (c *Client) WriteBytes(ctx context.Context, name string, data []byte) (int64, error) {
 	ue := uploadinfo.EntryFromBlob(data)
 	ch, err := chunker.New(ue, false, int(c.ChunkMaxSize))
 	if err != nil {
-		return err
+		return 0, err
 	}
 	return c.WriteChunked(ctx, name, ch)
 }
 
 // WriteChunked uploads chunked data with a given resource name to the CAS.
-func (c *Client) WriteChunked(ctx context.Context, name string, ch *chunker.Chunker) error {
+func (c *Client) WriteChunked(ctx context.Context, name string, ch *chunker.Chunker) (int64, error) {
```

Comment:
Actually, now that I think about it, I'm really not sure this function needs to be public at all. It's very low-level. But I suggest a separate CL only to make this one private (so that it can easily be rolled back if it breaks someone external).
This way, we only change the signature of UploadIfMissing.

Reply:
Separate CL it is!